pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -903,15 +903,15 @@ func (pl *dynamicResources) Filter(ctx context.Context, cs *framework.CycleState
 	if !pl.enabled {
 		return nil
 	}
+	logger := klog.FromContext(ctx)
```

Comment:
Let's not do this. It has the drawback that `klog.FromContext` always gets called, whether there are errors nor not and regardless whether the plugin needs to do something.

The way it was written before makes the code a bit larger, but in most cases (= no error, plugin inactive) the performance is slightly better.

Reply:
OK, got it. Because I saw that the initialization method was a little different in some places, so I tried to modify it.
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go#L713
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go#L1318

Reply:
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go#L713 is different because in the common path the error is not reached and a logger is always needed for the `logger.V(5).Info("pod resource claims"`.